codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
> We do this in all other places such as our Config class wherein if the user explicitly overrides the browser flavor to an empty value, we try to use to and then throw an error no ? Why this change of behavior here ?

`Config` is not presented as or part of a maven plugin

> Another example of wherein we are going against the Maven conventions.
The maven surefire plugin lets us define the suite xml location via a properties. If I explicitly default it to an empty value, I guess I see a NullPointerException or something on those lines.

I don't buy this argument. The `NullPointerException` in this case is because there is no acceptable default for surefire to figure out or convention that they could have coded (other than search the source tree for a testng suite file -- which seems overzealous) for it to fallback on. However, if the user defined a default `<property>` `<suiteXmlFile>` in the POM, that condition goes away too. 

> So IMO we should use defaults if no values were provided. If values were provided we attempt to validate the values.

agree, agree,

>If the validation fails, throw an error and bail out.

 disagree for this scenario.. Especially since omitting a `package` declaration in this situation solves the problem AND is more in line with what you would expect -- compared to the eclipse and intelij scenarios. So, I'm leaning in the direction that the right place to solve this is in the velocity template .. if `$package` is empty then do not write the  `package $package;` line

>  If the values were valid, we just use them and move ahead.

agree